### DISCUSSION TITLE
Is it possible to get modifier key status when a button clicked?

### DISCUSSION THREAD

**Discussion by aztack** [fyne-io/fyne#3351] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I need to get modifer keys (cmd/shift/ctrl/alt ) status when mouse click a button

### Describe the solution you'd like to see.

In a web page, it is very convenient to get modifier key status:
```js
document.body.addEventListener('click', e => {
  if (e.shiftKey) {
    // do one thing if button clicked with holding the shift key
  } else {
    // do another thing
  }
});
```
Is is possible in fyne?
```go
button := widget.NewButton("OK", func (event fyne.KeyEvent) {
  if event.ShiftKey {
   // do one thing if button clicked with holding the shift key
  } else {
    // do another thing
  }
})
```

Or is there another way to get modifier key status?

**Comment by Jacalz** [comment]
Please use discussions for questions, not issues. I will transfer this to the discussions page.

**Comment by andydotxyz** [comment, marked as answer]
What you post as equivalent web code actually isn't. A button being tapped is not a click event - it is an abstraction of mouse/touch coded into the button widget.
If you listen to the actual mouse events they have a modifier set just like on the web https://developer.fyne.io/api/v2.1/driver/desktop/mouseable.html. Of course that code is desktop specific, as a touchscreen does not work that way.

**Reply by aztack** [reply]
Could you give me a hint on how to listen mouse event? I can not find a demo/snippets on this.

**Reply by Jacalz** [reply]
I would suggest that you have a look at my presentation from FyneConf this year: https://m.youtube.com/watch?v=TDae1Kmxu_0. It goes through how you can use interfaces to add functionality to widgets, just like you would do in this case
